### ISSUE TITLE
Custom Square component

### ISSUE THREAD

**Issue by leedavidcs** [Clariity/react-chessboard#59]
I would like to apply additional elements within a square, based on which square it is. Would it be possible to support a custom square renderer?

Currently, I see `Square` renders a div that takes a `ref`, `style` and `children`: https://github.com/Clariity/react-chessboard/blob/main/src/chessboard/components/Square.tsx#L123-L132

Instead of this `div`, would it be possible to provide our own renderer with this type?
```ts
interface CustomSquareProps {
    // pass down the children
    children: ReactNode;
    // pass down the ref
    squareRef: RefObject<HTMLElement>;
    // pass down the style
    style: CSSProperties;

    // extra props
    square: Square;
    squareColor: "white" | "black";
}

type CustomSquareRenderer = FC<CustomSquareProps>;
```

**Comment by Clariity** [comment]
Sounds like a nice idea, can you create a PR for it and add an example story to storybook to show it's use and code?

**Comment by leedavidcs** [comment]
Ah ofc, certainly. I'll get around to making a PR in a few days :ok_hand:

**Comment by Clariity** [comment]
solved #60
